test/integration/default/controls/xcode.rb

```diff
@@ -35,3 +30,24 @@
     end
   end
 end
+
+control 'xcode-beta' do
+  title 'beta integrated development environment for macOS'
+  desc '
+    Verify that Xcode beta exists, developer mode is enabled, and the expected
+    simulators are installed using the xcversion commandline utility
+  '
+
+  macos_version = command('/usr/bin/sw_vers -productVersion').stdout.strip
+
+  describe file('/Applications/Xcode.app') do
+    it { should exist }
+    it { should be_symlink }
+  end
+
+  if macos_version.match? Regexp.union ['10.12', '10.13']
```

Comment:
Nit pick: the brackets are not necessary `Regexp.union '10.12', '10.13'` will work the same way.